### PR TITLE
Fixing JSON deserialization and adding HTTP2 support.

### PR DESCRIPTION
**Pull request by aloisdeniel**

### CODE REVIEW

**Comment by aloisdeniel** [comment]
The CI seems to failed and I'm not sure that environment variables are set for them.

File: lib/src/models/style_type.dart [hunk 1]
```diff
@@ -11,3 +11,15 @@ enum StyleType {
   @JsonValue('GRID')
   grid
 }
+
+/// The type of style as string enum for keys
+enum StyleTypeKey {
```
Comment (arnemolland) [inline lib/src/models/style_type.dart:16]: There should be a `stroke` value as well

**Review by arnemolland — COMMENTED** [review]
Looks great, thanks a lot for all the work being done here! I found an extra parameter in the API spec that should be included as well.

**Comment by arnemolland** [comment]
I'm taking a look at the environment (at the moment refs to private Figma files, these should be open/available in the future).

File: lib/src/client.dart [hunk 2]
```diff
@@ -101,7 +102,7 @@ class FigmaClient {
       [FigmaQuery query]) async {
     final uri = Uri.https(base, '$apiVersion$path', query?.params);
 
-    return await get(uri.toString(), headers: _authHeaders).then((res) {
+    return await await _send('GET', uri, _authHeaders).then((res) {
```
Comment (arnemolland) [inline lib/src/client.dart:105]: I would guess the double await is a typo?

File: lib/src/client.dart [hunk 3]
```diff
@@ -135,8 +135,61 @@ class FigmaClient {
...
+        uri.host,
+        uri.port,
+        supportedProtocols: ['h2'],
+      ),
+    );
+
+    var stream = transport.makeRequest(
+      [
+        Header.ascii(':method', method),
+        Header.ascii(':path', uri.path),
```
Comment (arnemolland) [inline lib/src/client.dart:151]: The `path` property does not include the query parameters, these will need to be added as well, e.g.
```dart
Header.ascii(':path', uri.path + (uri.hasQuery ? '?${uri.query}' : '')),
```

**Comment by arnemolland** [comment]
The environment looks good now, so the requested changes should give us a green build 😁 Again, thanks a lot!

File: lib/src/models/style_type.dart (hunk 1 above)
Comment (aloisdeniel) [reply lib/src/models/style_type.dart:16]: Are you sure about that ?

From the documentation there is no `STROKE` definition.

image

**Comment by aloisdeniel** [comment]
Okay, I implemented the fixes but it seems that environment variables are still missing on CI.

File: lib/src/models/style_type.dart (hunk 1 above)
Comment (arnemolland) [reply lib/src/models/style_type.dart:16]: Yes, the Figma documentation is lacking unfortunately..

**Comment by arnemolland** [comment]
Looks great! All tests passed locally when rebased on master, so I'll just merge this to a temp branch to get the build green before merging into master. Thanks again for the great work!

**Pull request merged**